src/datomic_toolbox/core.clj

```diff
@@ -21,97 +20,28 @@
 
 (defn db [] (d/db (connection)))
 
+(defn tempid
+  ([]  (d/tempid (partition)))
+  ([n] (d/tempid (partition) n)))
+
 (defn transact [tx-data]
```

Comment:
@tie-rack Yeah, that makes sense, but let me step back a bit first. "Having multiple arities of the same function scattered across namespaces" isn't and was never the intent of the design here.

It was/is: "There are convenient functions in the `core` namespace that allow for stateful configuration of a Datomic peer and they are implemented in terms of lower-level functions in other namespaces that are a) not stateful and b) free for your use as well." So they are not the "same functions" and that's not the way we should be thinking about them.

The questions on the table right now are:

1. Should we change the fundamental architecture of this branch in any way?
1. How should we handle `swap-tx!` and `deftx-data-fn` that are currently (sometimes) relying on stateful config but do not live in the `core` namespace? (This one could be rendered moot depending on what we decide for no. 1 above.)